compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/providers/impl/FirTypeResolverImpl.kt

```diff
@@ -92,14 +96,34 @@ class FirTypeResolverImpl(private val session: FirSession) : FirTypeResolver() {
         }
     }
 
+    private fun FirQualifierResolver.resolveEnumSymbol(
+        qualifier: List<FirQualifierPart>,
+        classId: ClassId
+    ): FirVariableSymbol<FirEnumEntry>? {
+        val enumClassSymbol = resolveSymbolWithPrefix(qualifier.dropLast(1), classId) ?: return null
+        val enumClassFir = enumClassSymbol.fir as? FirRegularClass ?: return null
+        if (enumClassFir.classKind != ClassKind.ENUM_CLASS) return null
+        return (enumClassFir.declarations.firstOrNull { it is FirEnumEntry && it.name == qualifier.last().name } as? FirEnumEntry)?.symbol
```

Comment:
And here, we use that last part of qualifier to pick the right enum entry?

Also, `resolveEnumEntrySymbol` as function name?

Reply:
Yep, let me introduce a local variable to make this more explicit.